system/logaccess/Grafana/CurlClient/GrafanaCurlClient.cpp

```diff
@@ -230,18 +263,54 @@ void formatResultLine(StringBuffer & returnbuf, const char * resultLine, const c
 
 /*
  * This method consumes an Iterator of values elements from a successful Grafana Loki query
- * It ignores the 1st child (ingest timestamp in ns), and formats the 2nd child (log line) into the desired format
+ * It extracts the appropriate "stream" values based on return column mode, and the values' 1st and 2nd children
+ * which represent timestamp in ns, and the log line, and formats into the requested format
  */
-void processValues(StringBuffer & returnbuf, IPropertyTreeIterator * valuesIter, LogAccessLogFormat format, bool & isFirstLine)
+void GrafanaLogAccessCurlClient::processValues(StringBuffer & returnbuf, IPropertyTreeIterator * valuesIter, IPropertyTree * stream, LogAccessLogFormat format, const LogAccessReturnColsMode retcolmode, bool & isFirstLine)
 {
+    if (!valuesIter)
+        return;
+
+    Owned<IProperties> fieldValues = createProperties(true);
+
+    //extract the requested fields from the stream if it's available
+    if (stream)
+    {
+        switch (retcolmode)
+        {
+        case RETURNCOLS_MODE_all:
+        {
+            fieldValues->setProp(m_nodeColumn.name, stream->queryProp(m_nodeColumn.name));
+            fieldValues->setProp(m_containerColumn.name, stream->queryProp(m_containerColumn.name));
+            fieldValues->setProp(m_instanceColumn.name, stream->queryProp(m_instanceColumn.name));
+            [[fallthrough]];
```

Comment:
C17 feature, not sure if there's any concern using c17 specific constructs

Reply:
Nice, I hadn't picked up on that feature.  We require c++17, so no problem using the feature  - it has the advantage it will be checked by the compiler.